app/Actions/Diagnostics/Checks/IniSettingsCheck.php

```diff
@@ -36,62 +37,47 @@ public function check(array &$errors): void
+		if ($this->convert_size(ini_get('post_max_size')) < $this->convert_size('100M')) {
+			$errors[] = 'Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.';
 		}
 		$max_execution_time = intval(ini_get('max_execution_time'));
 		if (0 < $max_execution_time && $max_execution_time < 200) {
-			$errors[]
-				= 'Warning: You may experience problems when uploading a photo of large size or handling many/large albums. Take a look in the FAQ for details.';
+			$errors[] = 'Warning: You may experience problems when uploading a photo of large size or handling many/large albums. Take a look in the FAQ for details.';
 		}
 		if (empty(ini_get('allow_url_fopen'))) {
-			$errors[]
-				= 'Warning: You may experience problems with the Dropbox- and URL-Import. Edit your php.ini and set allow_url_fopen to 1.';
+			$errors[] = 'Warning: You may experience problems with the Dropbox- and URL-Import. Edit your php.ini and set allow_url_fopen to 1.';
 		}
 
 		// Check imagick
 		if (!extension_loaded('imagick')) {
-			$errors[]
-				= 'Warning: Pictures that are rotated lose their metadata! Please install Imagick to avoid that.';
+			$errors[] = 'Warning: Pictures that are rotated lose their metadata! Please install Imagick to avoid that.';
 		} else {
 			if (!isset($settings['imagick'])) {
-				$errors[]
-					= 'Warning: Pictures that are rotated lose their metadata! Please enable Imagick in settings to avoid that.';
+				$errors[] = 'Warning: Pictures that are rotated lose their metadata! Please enable Imagick in settings to avoid that.';
 			}
 		}
 
 		if (!function_exists('exec')) {
-			$errors[]
-				= 'Warning: exec function has been disabled. You may experience some error 500, please report them to us.';
+			$errors[] = 'Warning: exec function has been disabled. You may experience some error 500, please report them to us.';
 		}
 
 		if (empty(ini_get('user_agent'))) {
-			$errors[]
-				= 'Warning: user_agent for PHP is not set. You may experience problems when importing images via URL.';
+			$errors[] = 'Warning: user_agent for PHP is not set. You may experience problems when importing images via URL.';
 		}
 
-		if (!ini_get('assert.exception')) {
-			$errors[]
-				= 'Warning: assert.exception is set to false. Lychee assumes that failing assertions throw proper exceptions.';
+		if (!(bool) ini_get('assert.exception')) {
+			$errors[] = 'Warning: assert.exception is set to false. Lychee assumes that failing assertions throw proper exceptions.';
 		}
 
-		if (ini_get('zend.assertions') != -1 && !config('app.debug')) {
-			$errors[]
-				= 'Warning: zend.assertions is enabled although Lychee is not in debug mode. Outside of debugging, code generation for assertions is recommended to be disabled for efficiency reasons';
+		if (ini_get('zend.assertions') !== '-1' && !(bool) config('app.debug')) {
```

Comment:
> By the way, what happens if users set APP_DEBUG = false in their .env file? Does Laravel somehow normalize these variants for us? I somehow assume so, because otherwise we would already have seen more bug reports.

This normalization is done if config/app.php

Reply:
That's good. But it stills leaves the question open, if we want to avoid boolean casts.

Reply:
I implemented a suggestion myself. All casts and conversions to boolean are removed by https://github.com/LycheeOrg/Lychee/pull/1336/commits/3a4b8f778b81b5a3582f855d1fc9673f90e0cc28. On top, https://github.com/LycheeOrg/Lychee/pull/1336/commits/996253e4ab8d4b9b197bd9741eb0ffedd144171d enforces this. We only need to discuss if we want it. @kamil4? @ildyria?

Reply:
I think we are clear there.